### DISCUSSION TITLE
[Q&A] Remote inventory by SSH through a PAM system (Privileged Access Manager)

### DISCUSSION THREAD

**Discussion by alexmicontini** [glpi-project/glpi-agent#436] [Q&A]
### Documentation & bug reporting acknowledgment

Yes, I read it

### Describe your problem

Hi,

in my organization we have about 300 linux virtual machines that we need to inventory. The inventory we want to do through the remote inventory task of the glpi agent, not by installing the agent.

The remote inventory task for Linux operating systems uses the SSH protocol, both by username and password and by public key.

The problem we find ourselves with is that we want the SSH connection to be made through a PAM (privileged access management) system, like the one used by companies:

- Broadcom: https://techdocs.broadcom.com/us/en/symantec-security-software/identity-security/identity-management-and-governance-connectors/1-0/connectors/ca-connectors/ca-privileged-access-manager.html
- Delinea: https://delinea.com/es/products/secret-server

to have a key rotation and a record of the connections.

Is it possible for the glpi agent to use a PAM system for the SSH connection to the remote servers?

GLPI version: 10.0.7
GLPI Agent Version: 1.5

Thank you so much. All the best

**Comment by g-bougard** [comment]
Hello @alexmicontini 

at the moment, the only possible way is to manage the remoteinventory task from a script which export USERNAME & PASSWORD environment variables or export the private key to use in an expected place. Does Broadcom or Delinea permits to export such information from a script ?

**Reply by alexmicontini** [reply]
Hi @g-bougard,

Thank you very much for answering.

I will ask our PAM system partner that we have to see if it is possible to export such information from a script, as long as they tell me I will reply.

Greetings
